Assets/Game/Addons/ModSupport/ModManager.cs

```diff
@@ -948,27 +948,36 @@ public static string MakeRelativePath(string path0, string path1)
         /// This method replicates the same behaviour for mods, doing all the hard work at build time.
         /// Results are stored to json manifest file for performant queries at runtime.
         /// </remarks>
-        public static void SeekModContributes(ModInfo modInfo)
+        public static void SeekModContributes(ModInfo modInfo, bool automaticallyRegisterQuestLists)
         {
             List<string> spellIcons = null;
             List<string> booksMapping = null;
+            List<string> questLists = null;
 
-            foreach (string file in modInfo.Files)
+            foreach (var file in modInfo.Files)
             {
-                string directory = Path.GetDirectoryName(file);
+                var directory = Path.GetDirectoryName(file);
 
-                if (directory.EndsWith("SpellIcons"))
+                if (directory != null && directory.EndsWith("SpellIcons"))
```

Comment:
For strings, it makes more sense to check `string.IsNullOrEmpty(directory)`. Especially since Path.GetDirectoryName is specified to return an empty string sometimes: https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getdirectoryname